### ISSUE TITLE
Standardize signing serialization `offchain-data-wrapper`

### ISSUE THREAD

**Issue by eelanagaraj** [celo-org/celo-monorepo#8862]
With [this PR](https://github.com/celo-org/celo-monorepo/pull/8836), the current implementation of `offchain-data-wrapper` is slightly less efficient than before, but should not suffer from the same edge case that this PR fixes (except in the case that the guessedSignature must still be used, which seems unavoidable). (See that PR description for more details).

We should long-term move towards one type of signature serialization (either VRS or RSV) throughout the module. Known places where this needs to be standardized:
- in `PublicSimpleAccessor.sign` we are actually using the `wallet.signTypedData` method which returns an RSV-signature (we can switch this to using the `CK.signTypedData` + `SignatureUtils.serializeSignature` to make this VRS), but this increases the scope of this PR)
we could just use the RSV recover function, but this moves away from compatibility with `serializeSignature` (VRS)
- in tests for this module, `serializeSignature` is used
- ?? -> carefully go through module and check where any signing/sig-parsing happens

Once all changes have been made, we can reduce the calls to `verifyEIP712TypedDataSigner` and instead call either `recoverEIP712TypedDataSignerRsv` or `recoverEIP712TypedDataSignerVrs` once.

Feel free to reach out to @eelanagaraj for more context!

**Comment by aslawson** [comment]
@eelanagaraj is the suggestion for the applications team to standardize the signature schema with other use cases (eg. offchain storage?)

>  slightly less efficient than before, should not suffer from the same edge case that this PR fixes
what is the impact of this inefficiency?

**Comment by eelanagaraj** [comment]
hey @aslawson ! _Yes sort of_, additional context here: @nambrot and I chatted about this a bit -- since the offchain storage use case is fairly new, we have the opportunity to standardize usage of one signature schema (i.e. VRS xor RSV). 

The _sort of_: standardizing across CK/monorepo would be nice but it may be too late for this to some extent, in that it would break a ton of downstream apps using one or the other I imagine. Even if we want to do that, it's likely a sizeable chunk of work IMO and possibly out of scope for applications. (Examples: would involve modifying all the `wallet` serializations or CK `signTypedData` + `serializeSignature`, + fixing all dependent uses.) That's why I limited this ticket in scope to just standardizing signing within the `offchain-data-wrapper` module, which at least isn't actively in use from what I understand.

"slightly less efficient" in this case (from very rough benchmarking) is on the order of a few milliseconds in total

Lmk if anything else needs to be clarified!
